Source/WebCore/platform/libwpe/PlatformKeyboardEventLibWPE.cpp

```diff
@@ -758,6 +758,12 @@ String PlatformKeyboardEvent::keyCodeForHardwareKeyCode(unsigned keyCode)
         return "F24"_s;
     case 0x00E1:
         return "BrowserSearch"_s;
+    case 0x00E7:
+        return "Cancel"_s;
+    case 0x00F8:
+        return "Unknown"_s;
+    case 0x016B:
+        return "Clear"_s;
```

Comment:
Where do these codes come from? I have not been able to find them in the MDN pages linked from the comments in this file.

Reply:
You just take the corresponding [input event key code](https://github.com/freedesktop/libevdev/blob/libevdev-1.13.0/include/linux/linux/input-event-codes.h) and [add 8 to it](https://github.com/Igalia/cog/blob/0.16.1/platform/wayland/cog-platform-wl.c#L1212-L1214) to get an equivalent MDN key code. These new key codes are not currently listed in the MDN pages so I just named them based off of the corresponding key event code define names.

Reply:
Ah, I saw now your comment [in bug #251778](https://bugs.webkit.org/show_bug.cgi?id=251778)... Not sure how much we can we expect there will be other devices which use the same codes. Also I wonder why your device would not produce `WPE_KEY_Cancel` instead of `0x00E7` 🤔

Reply:
> Not sure how much we can we expect there will be other devices which use the same codes. 

They are kernel defined key codes I guess so I presume there are other devices using them.

> Also I wonder why your device would not produce `WPE_KEY_Cancel` instead of `0x00E7` 🤔

It produces both, [`WPE_KEY_Cancel`](https://github.com/WebPlatformForEmbedded/libwpe/blob/f4cbc1ff330d31f8761585bd392e8d093257c0bc/include/wpe/keysyms.h#L202) is the keysym not the hardware key code:
```
handle_key_event key: 0x000000e7 keysym: 0x0000ff69
```